core/Pos/Crypto/SecretSharing.hs

```diff
@@ -120,77 +125,143 @@ instance Bi SecretProof =>
+reorderDecryptedShares
+    :: [(VssPublicKey, Int)]            -- ^ Participants
+    -> HashMap VssPublicKey [DecShare]  -- ^ Decrypted shares
+    -> [(Scrape.ShareId, DecShare)]
+reorderDecryptedShares participants shares =
+    map (first toInteger) $ go 1 participants
   where
-    convert (EncShare es, VssPublicKey p, Share s) = (es, p, s)
+    go :: Int                        -- ^ Index of current share
+       -> [(VssPublicKey, Int)]      -- ^ Remaining participants
+       -> [(Int, DecShare)]
+    go _ []         = []
+    go i ((k,n):ps) = case HM.lookup k shares of
+        Nothing -> go (i + n) ps
+        Just ss -> zip [i..] (take n ss) ++ go (i + n) ps
 
 -- CHECK: @verifyEncShare
--- | Verify an encrypted share using SecretSharingExtra.
-verifyEncShare :: SecretSharingExtra -> VssPublicKey -> EncShare -> Bool
-verifyEncShare (SecretSharingExtra e c) (VssPublicKey pk) (EncShare es) =
-    Pvss.verifyEncryptedShare e c (es, pk)
+-- | Verify encrypted shares
+verifyEncShares
+    :: MonadRandom m
+    => SecretProof
+    -> Scrape.Threshold
+    -> [(VssPublicKey, EncShare)]
+    -> m Bool
+verifyEncShares SecretProof{..} threshold (sortWith fst -> pairs)
+    | threshold <= 1     = error "verifyEncShares: threshold must be > 1"
+    | threshold >= n - 1 = error "verifyEncShares: threshold must be < n-1"
+    | otherwise =
+          Scrape.verifyEncryptedShares
+              spExtraGen
+              threshold
+              spCommitments
+              spParallelProofs
+              (coerce $ map snd pairs)  -- shares
+              (coerce $ map fst pairs)  -- participants
+  where
+    n = fromIntegral (length pairs)
 
 -- CHECK: @verifyShare
--- | Verify that Share has been decrypted correctly.
-verifyShare :: EncShare -> VssPublicKey -> Share -> Bool
-verifyShare (EncShare es) (VssPublicKey pk) (Share sh) =
-    Pvss.verifyDecryptedShare (es, pk, sh)
+-- | Verify that DecShare has been decrypted correctly.
+verifyDecShare :: VssPublicKey -> EncShare -> DecShare -> Bool
+verifyDecShare (VssPublicKey pk) (EncShare es) (DecShare sh) =
+    Scrape.verifyDecryptedShare (es, pk, sh)
 
 -- CHECK: @verifySecretProof
 -- | Verify that SecretProof corresponds to Secret.
-verifySecretProof :: SecretSharingExtra -> Secret -> SecretProof -> Bool
-verifySecretProof (SecretSharingExtra e c) (Secret s) (SecretProof p) =
-    Pvss.verifySecret e c s p
+verifySecret :: Scrape.Threshold -> SecretProof -> Secret -> Bool
+verifySecret thr SecretProof{..} (Secret secret) =
+    Scrape.verifySecret spExtraGen thr spCommitments secret spProof
+
+----------------------------------------------------------------------------
+-- Test
+----------------------------------------------------------------------------
+
+-- | You can use this to do debugging. If everything is okay with SCRAPE, it
+-- will print 'True's.
+testScrape
+    :: Int           -- ^ Threshold (number of participants = 2× threshold)
+    -> IO ()
+testScrape t = do
```

Comment:
No, it's because loading the test suite in repl is a pain (if you want to debug anything in scrape, after each change in the main project you hav to restart the repl).